bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/ContainerPostProcessor.java

```diff
@@ -0,0 +1,98 @@
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.services;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.servlets.post.Modification;
+import org.apache.sling.servlets.post.SlingPostProcessor;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.wcm.core.components.internal.models.v1.PanelContainerImpl;
+
+@Component(service = SlingPostProcessor.class,
+           property = {
+                   Constants.SERVICE_RANKING + ":Integer=" + Integer.MAX_VALUE
+           })
+public class ContainerPostProcessor implements SlingPostProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ContainerPostProcessor.class);
+    private static final String PARAM_ORDERED_CHILDREN = "itemOrder";
+    private static final String PARAM_DELETED_CHILDREN = "deletedItems";
+
+    @Override
+    public void process(SlingHttpServletRequest request, List<Modification> modifications) throws Exception {
+        ResourceResolver resourceResolver = request.getResource().getResourceResolver();
+        ArrayList<Modification> addedModifications = new ArrayList<Modification>();
+        if (accepts(request, resourceResolver)) {
+            Resource container = request.getResource();
+            try {
+                handleOrder(container, request);
+                handleDelete(container, request, resourceResolver, addedModifications);
+            } catch (RepositoryException e) {
+                LOGGER.error("Could not order items of the container at {}", container.getPath(), e);
+            }
+        }
+        modifications.addAll(addedModifications);
+    }
+
+    protected void handleOrder(Resource container, SlingHttpServletRequest request) throws RepositoryException {
+        String[] orderedChildrenNames = StringUtils.split(request.getParameter(PARAM_ORDERED_CHILDREN), ",");
+        if (orderedChildrenNames != null && orderedChildrenNames.length > 0) {
+            final Node containerNode = container.adaptTo(Node.class);
+            if (containerNode != null) {
+                for (int i = orderedChildrenNames.length - 1; i >= 0; i--) {
+                    if (i == orderedChildrenNames.length - 1 && containerNode.hasNode(orderedChildrenNames[i])) {
+                        containerNode.orderBefore(orderedChildrenNames[i], null);
+                    } else if (containerNode.hasNode(orderedChildrenNames[i]) && containerNode.hasNode(orderedChildrenNames[i])) {
```

Comment:
There's something fishy about the logic here, the condition `containerNode.hasNode(orderedChildrenNames[i]) && containerNode.hasNode(orderedChildrenNames[i])` looks wrong, I'm guessing the second index should be `[i+1]`. I would also restructure the `if` clauses to eliminate the common part:
```
if (containerNode.hasNode(orderedChildrenNames[i])) {
    if (i == orderedChildrenNames.length - 1) {
        ...
    } else if (containerNode.hasNode(orderedChildrenNames[i+1]) {
        ...
    }
}

Reply:
This logic was already in place in the old ContainerServlet https://github.com/adobe/aem-core-wcm-components/blob/main/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/ContainerServlet.java#L88

Reply:
updated logic in 8d44260f19687f5ebf8bdcce15351ac950aab624